Error for this location

https://localhost.api.flow.io/locations?address=Columbia+River,+Washington

returns a server error 

locally:

```
play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[RuntimeException: Could not find country from address [$address]]]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:265) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:191) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.api.GlobalSettings$class.onError(GlobalSettings.scala:179) [play_2.11-2.4.3.jar:2.4.3]
	at play.api.DefaultGlobal$.onError(GlobalSettings.scala:212) [play_2.11-2.4.3.jar:2.4.3]
	at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:94) [play_2.11-2.4.3.jar:2.4.3]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$9$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:158) [play-netty-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$9$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:155) [play-netty-server_2.11-2.4.3.jar:2.4.3]
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36) [scala-library-2.11.8.jar:na]
	at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:216) [scala-library-2.11.8.jar:na]
	at scala.util.Try$.apply(Try.scala:192) [scala-library-2.11.8.jar:na]
Caused by: java.lang.RuntimeException: Could not find country from address [$address]
	at scala.sys.package$.error(package.scala:27) ~[scala-library-2.11.8.jar:na]
	at utils.Google$$anonfun$parseResults$1.apply(Google.scala:92) ~[classes/:na]
	at utils.Google$$anonfun$parseResults$1.apply(Google.scala:82) ~[classes/:na]
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234) ~[scala-library-2.11.8.jar:na]
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234) ~[scala-library-2.11.8.jar:na]
	at scala.collection.immutable.List.foreach(List.scala:381) ~[scala-library-2.11.8.jar:na]
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234) ~[scala-library-2.11.8.jar:na]
	at scala.collection.immutable.List.map(List.scala:285) ~[scala-library-2.11.8.jar:na]
	at utils.Google.parseResults(Google.scala:82) ~[classes/:na]
	at utils.Google.getLocationsByAddress(Google.scala:76) ~[classes/:na]
```

--- a/api/app/utils/Google.scala
+++ b/api/app/utils/Google.scala
@@ -1,9 +1,10 @@
 package utils
 
-import io.flow.reference.Countries
-import io.flow.common.v0.models.Location
 import com.google.maps.{GeoApiContext, GeocodingApi}
 import com.google.maps.model.{AddressComponent, GeocodingResult}
+import io.flow.reference.Countries
+import io.flow.common.v0.models.Location
+import play.api.Logger
 
 object Google {
 
@@ -73,12 +74,19 @@ class Google @javax.inject.Inject() (
         Left(Seq(s"No results found for address: [$address]"))
       }
       case results => {
-        Right(parseResults(address, results))
+        Right(sortLocations(parseResults(address, results)))
       }
     }
   }
 
-  private def parseResults(address: String, results: Seq[GeocodingResult]): Seq[Location] = {
+  /**
+    * Ensures locations w/ countries are defined earlier in lost
+    */
+  private[this] def sortLocations(locations: Seq[Location]): Seq[Location] = {
+    locations.filter(_.country.isDefined) ++ locations.filter(_.country.isEmpty)
+  }
+
+  private[this] def parseResults(address: String, results: Seq[GeocodingResult]): Seq[Location] = {
     results.map { one =>
       val streetNumber = findAsString(one.addressComponents, Seq(Google.AddressComponentType.StreetNumber))
       val streetAddress = findAsString(one.addressComponents, Seq(Google.AddressComponentType.StreetAddress, Google.AddressComponentType.Route))
@@ -89,10 +97,22 @@ class Google @javax.inject.Inject() (
       val postal = findAsString(one.addressComponents, Seq(Google.AddressComponentType.PostalCode))
 
       val country = findAsString(one.addressComponents, Seq(Google.AddressComponentType.Country)) match {
-        case None => sys.error("Could not find country from address [$address]")
-        case Some(q) => Countries.find(q) match {
-          case None => sys.error(s"Reference lookup could not find country $q")
-          case Some(c) => Some(c.iso31663)
+        case None => {
+          Logger.warn(s"Could not determine country for address[$address]")
+          None
+        }
+
+        case Some(q) => {
+          Countries.find(q) match {
+            case None => {
+              Logger.warn(s"Country code[$q] was not valid in reference data. Skipping country")
+              None
+            }
+
+            case Some(c) => {
+              Some(c.iso31663)
+            }
+          }
         }
       }
 

--- a/api/app/utils/Google.scala
+++ b/api/app/utils/Google.scala
@@ -80,7 +80,7 @@ class Google @javax.inject.Inject() (
   }
 
   /**
-    * Ensures locations w/ countries are defined earlier in lost
+    * Ensures locations w/ countries are defined earlier in list
     */
   private[this] def sortLocations(locations: Seq[Location]): Seq[Location] = {
     locations.filter(_.country.isDefined) ++ locations.filter(_.country.isEmpty)

